app/javascript/components/home/HomePage.jsx

```diff
@@ -13,22 +16,70 @@ class HomePage extends Component {
         })
     }
 
-    renderCourseName = (course) => {
-        if (!course.can_read && !course.can_manage) {
-            return course.name;
-        } else if (course.can_read && !course.can_manage) {
-            return <a className={"list-group-item justify-content-between"} href={course.path}>{course.name}</a>;
-        } else if (course.can_manage) {
-            return <a className={"list-group-item justify-content-between list-group-item-info"}
-                      href={course.path}>{course.name}</a>;
-        }
-    }
-
     filterCourses = (query) => {
         const filteredCourses = this.state.courses.filter(c => c.name.indexOf(query) !== -1);
         this.setState({filteredCourses})
     }
 
+	renderCourseUrl = (cell, row) => <a href={row.path}>{cell}</a>;
+
+	renderCourseOrgLink = (cell, row) => {
+		const url = `https://github.com/${cell}`;
+		return (
+			<a href={url}>{cell}</a>
+		)
+	};
+
+    renderJoinButton = (cell, row) => {
+		return (
+            <div>
+                {!row.user_enrolled ?
+                    <Button className="btn btn-success" variant="success" data-testid={`join-button-${row.id}`} href={`${row.join_path}`} >Join</Button>
```

Comment:
This will result in a GET request rather than a POST request.  You'll need to change it to a POST, as well as addressing the CSRF Token.     I've put some pointers on the discussion of the pull request with hints how to approach this.